### PR TITLE
docs: release notes

### PR DESCRIPTION
**Pull request by rootulp**
Closes https://github.com/celestiaorg/celestia-app/issues/3791

### CODE REVIEW

File: RELEASES.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+### Library Consumers
+
+If you are a library consumer, a number of the Go APIs have changed since celestia-app v1.x.x. Some of the notable changes are:
+
+- Code pertaining to the original data square was extracted to [celestiaorg/go-square](https://github.com/celestiaorg/go-square).
+  - celestia-app v1.x had a shares package. celestia-app v2.x uses [go-square/shares](https://github.com/celestiaorg/go-square/tree/c8242f96a844956f8d1c60e5511104deed8bc361/shares)
+  - celestia-app v1.x had a blob.types package with `CreateCommitment`. celestia-app v2.x uses `CreateCommitment` from the [go-square/inclusion](https://github.com/celestiaorg/go-square/tree/c8242f96a844956f8d1c60e5511104deed8bc361/inclusion).
+- celestia-app v1.x had a lot of functionality included in the signer. celestia-app v2.x split a txClient from the signer.
```
Comment (staheri14) [inline RELEASES.md:20]: [nit]
```suggestion
- celestia-app v1.x had a lot of functionality included in the signer. celestia-app v2.x splits a txClient from the signer.
```

File: RELEASES.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+### Library Consumers
+
+If you are a library consumer, a number of the Go APIs have changed since celestia-app v1.x.x. Some of the notable changes are:
+
+- Code pertaining to the original data square was extracted to [celestiaorg/go-square](https://github.com/celestiaorg/go-square).
+  - celestia-app v1.x had a shares package. celestia-app v2.x uses [go-square/shares](https://github.com/celestiaorg/go-square/tree/c8242f96a844956f8d1c60e5511104deed8bc361/shares)
+  - celestia-app v1.x had a blob.types package with `CreateCommitment`. celestia-app v2.x uses `CreateCommitment` from the [go-square/inclusion](https://github.com/celestiaorg/go-square/tree/c8242f96a844956f8d1c60e5511104deed8bc361/inclusion).
```
Comment (staheri14) [inline RELEASES.md:19]: [nit]
```suggestion
  - celestia-app v1.x had a blob.types package with `CreateCommitment` function. celestia-app v2.x uses `CreateCommitment` function from the [go-square/inclusion](https://github.com/celestiaorg/go-square/tree/c8242f96a844956f8d1c60e5511104deed8bc361/inclusion).
```

File: RELEASES.md (hunk 1 above)
Comment (staheri14) [inline RELEASES.md:20]: >celestia-app v1.x had a lot of functionality included in the signer. celestia-app v2.x split a txClient from the signer.

Can a link be given for this part?

File: RELEASES.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+- celestia-app v1.x had a lot of functionality included in the signer. celestia-app v2.x split a txClient from the signer.
+
+## v3.0.0
```
Comment (staheri14) [inline RELEASES.md:22]: Is this section finalized? if not, it might be a good idea to indicate it somehow, like "in progress", "draft", etc.

**Review by staheri14 — COMMENTED** [review]
LGTM!, left some comments.

File: RELEASES.md (hunk 3 above)
Comment (rootulp) [reply RELEASES.md]: https://github.com/celestiaorg/celestia-app/pull/3794/commits/253c613798bf5395f1aad3e62f2aabb362729816

File: RELEASES.md (hunk 1 above)
Comment (rootulp) [reply RELEASES.md]: https://github.com/celestiaorg/celestia-app/pull/3794/commits/253c613798bf5395f1aad3e62f2aabb362729816

**Review by staheri14 — DISMISSED** [review]
I can't speak to whether every single detail required for the upgrade is included, but from my perspective as a general reader, it looks good to me!

**Review by evan-forbes — DISMISSED** [review]

File: RELEASES.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+- Code pertaining to the original data square was extracted to [celestiaorg/go-square](https://github.com/celestiaorg/go-square).
+  - celestia-app v1.x had a shares package. celestia-app v2.x uses [go-square/shares](https://github.com/celestiaorg/go-square/tree/c8242f96a844956f8d1c60e5511104deed8bc361/shares)
+  - celestia-app v1.x had a blob.types package with `CreateCommitment` function. celestia-app v2.x uses `CreateCommitment` function from the [go-square/inclusion](https://github.com/celestiaorg/go-square/tree/c8242f96a844956f8d1c60e5511104deed8bc361/inclusion).
+- celestia-app v1.x had a lot of functionality included in the signer. celestia-app v2.x splits a txClient from the signer. See [#3433](https://github.com/celestiaorg/celestia-app/pull/3433).
+
+## v3.0.0 [draft]
+
+### Node operators
+
+- Consensus node operators should enable the BBR (Bottleneck Bandwidth and Round-trip propagation time) congestion control algorithm. See [#3774](https://github.com/celestiaorg/celestia-app/pull/3774).
```
Comment (cmwaters) [inline RELEASES.md:26]: Don't we want this mentioned in v2?

File: RELEASES.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+### Library Consumers
+
+If you are a library consumer, a number of the Go APIs have changed since celestia-app v1.x.x. Some of the notable changes are:
+
+- Code pertaining to the original data square was extracted to [celestiaorg/go-square](https://github.com/celestiaorg/go-square).
+  - celestia-app v1.x had a shares package. celestia-app v2.x uses [go-square/shares](https://github.com/celestiaorg/go-square/tree/c8242f96a844956f8d1c60e5511104deed8bc361/shares)
+  - celestia-app v1.x had a blob.types package with `CreateCommitment` function. celestia-app v2.x uses `CreateCommitment` function from the [go-square/inclusion](https://github.com/celestiaorg/go-square/tree/c8242f96a844956f8d1c60e5511104deed8bc361/inclusion).
+- celestia-app v1.x had a lot of functionality included in the signer. celestia-app v2.x splits a txClient from the signer. See [#3433](https://github.com/celestiaorg/celestia-app/pull/3433).
+
+## v3.0.0 [draft]
```
Comment (cmwaters) [inline RELEASES.md:22]: nit:
```suggestion
## v3.0.0 [Unreleased]
```

**Review by cmwaters — COMMENTED** [review]
Not if this is your plan, but I think the latest release should always be at the top. Also I'm not sure if we should keep the pending release information in the same document. I don't want to confuse users about releases that aren't ready

File: RELEASES.md (hunk 4 above)
Comment (rootulp) [reply RELEASES.md:26]: I think it's optional in v2 and required in v3 so I chose to document it in v3. We don't have the warning yet in v2 so I don't think we need to document in in the v2 section.

When we add the warning to v2, we can also copy a variant of this that says "optional" to the v2 section.

Comment (rootulp) [reply RELEASES.md]: cc: @evan-forbes for thoughts

**Review by staheri14 — DISMISSED** [review]

File: RELEASES.md (hunk 4 above)
Comment (cmwaters) [reply RELEASES.md]: I would inform users in v2. We want all of them to upgrade in v2

Comment (rootulp) [reply RELEASES.md]: Can do. Then this PR is blocked on the PR with a warning and instructions on how to enable BBR so some variant of https://github.com/celestiaorg/celestia-app/pull/3774

**Comment by rootulp** [comment]
This feels a bit duplicative of the information in the actual v2.0.0 release notes. The only thing missing from the v2.0.0 release notes that is present here is the note to enable BBR which wasn't actually included in the v2.0.0 release.

File: docs/release-notes/release-notes.md [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+
+## [v2.0.0](https://github.com/celestiaorg/celestia-app/releases/tag/v2.0.0)
+
+### Node Operators
+
+If you are a consensus node operator, please follow the communication channels listed under [network upgrades](https://docs.celestia.org/nodes/participate#network-upgrades) to learn when this release is recommended for each network (e.g. Mocha, Mainnet Beta).
+
+Consensus node operators are expected to upgrade to this release _prior_ to the Lemongrass hardfork if they intend to continue participating in the network. The command used to start the [consensus node](https://docs.celestia.org/nodes/consensus-node#start-the-consensus-node) or [validator node](https://docs.celestia.org/nodes/validator-node#run-the-validator-node) will accept an additional `--v2-upgrade-height` flag. See [this table](https://docs.celestia.org/nodes/hardfork-process#lemongrass-hardfork) for upgrade heights for each network.
+
+Consensus node operators should enable the BBR (Bottleneck Bandwidth and Round-trip propagation time) congestion control algorithm. See [#3812](https://github.com/celestiaorg/celestia-app/pull/3812).
```
Comment (rootulp) [inline docs/release-notes/release-notes.md:13]: FYI: @cmwaters

**Review by cmwaters — APPROVED** [review]

**Review by ninabarbakadze — APPROVED** [review]

**Pull request merged**